package_bag/routines.py

```diff
@@ -0,0 +1,211 @@
+        has_rights = []
+        for bag in Bag.objects.filter(rights_data__isnull=True):
+            try:
+                rights_json = self.retrieve_rights(bag, url, apikey)
+                self.save_rights(bag, rights_json)
+                has_rights.append(bag)
+            except Exception as e:
+                print(e)
+        # get rights ids from database
+        # loop through rights ids
+        # retrieve rights
+        # save rights
+        return has_rights
+
+    def retrieve_rights(self, bag, url, apikey):
+        """Sends POST request to rights statement service, receives JSON in return"""
+        # url for get request
+        resp = post(
+            url,
+            data="data",  # TO DO: what data is sent to rights service? obviously includes rights ids
+            headers={
+                "Content-Type": "application/json",
+                "apikey": apikey,
+            },
+        )
+        # send get request
+        # get serialized rights back as json
+        # QUESTION: do we want to validate the json we get back?
+        # return saved json
+        return rights_json
+
+    def save_rights(self, bag, rights_json):
+        """Save JSON from rights statement service to database"""
+        bag.rights_data = rights_json
+        bag.save()
+        # save json...to file? rights.json?
+        pass
+
+
+class CreatePackage(object):
+    """Create JSON according to Ursa Major schema and package with bag"""
+
+    def run(self):
+        temp_dir = settings.TMP_DIR
+        dest_dir = settings.DEST_DIR
+        packaged = []
+        unpackaged = Bag.objects.filter(rights_data__isnull=False)
+        for u in unpackaged:
+            try:
+                self.create_json(u, temp_dir)
+                self.package_bag(temp_dir, dest_dir, u)
+                packaged.append(u)
+            except Exception as e:
+                print(e)
+        return packaged
+
+    def create_json(self, bag, temp_dir):
+        """Create JSON file to send to Ursa Major"""
+        bag_json = BagSerializer(bag).data
+        # print(bag_json)
+        with open(join(temp_dir, bag.bag_identifier, "{}.json".format(bag.bag_identifier),), "w",) as f:
+            json.dump(bag_json, f, indent=4, sort_keys=True, default=str)
+        return join(temp_dir, bag.bag_identifier,
+                    "{}.json".format(bag.bag_identifier))
+
+    def package_bag(self, temp_dir, dest_dir, bag):
+        """Create package to send to Ursa Major"""
+        tar_filename = "{}.tar.gz".format(bag.bag_identifier)
+        with tarfile.open(join(temp_dir, tar_filename), "w:gz") as tar:
+            tar.add(join(temp_dir, bag.bag_identifier),
+                    arcname=basename(join(temp_dir, bag.bag_identifier)))
+        mkdir(
+            join(dest_dir, bag.bag_identifier)
+        )
+        move(
+            join(temp_dir, tar_filename),
+            join(dest_dir, bag.bag_identifier, tar_filename,),
+        )
```

Comment:
I would consider breaking this into two functions, which would allow you to again think more carefully about what you need to pass into each of them.